### PR TITLE
fix: allow saving ai generated content type

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

- Just hardcodes some schema values to ensure the schema is saveable

### CODE REVIEW

File: packages/core/content-type-builder/server/src/services/architect.ts [hunk 1]
```diff
@@ -151,12 +152,46 @@ const systemPrompt = `
           </examples>
     `;
 
+const applyDefaultValues = (schema: any) => {
```
Comment (Marc-Roig) [inline packages/core/content-type-builder/server/src/services/architect.ts:155]: I am trying to fine tune the prompt but this seems like a good failsafe

File: packages/core/content-type-builder/server/src/services/architect.ts [hunk 2]
```diff
@@ -31,11 +31,12 @@ const systemPrompt = `
             </schema_kind>
 
             <info_properties>
+              You MUST provide all the following properties:
+              - name: Same as pluralName
```
Comment (Convly) [inline packages/core/content-type-builder/server/src/services/architect.ts:35]: Does it actually work? I've tried similar prompts but didn't manage to fix the issue

File: packages/core/content-type-builder/server/src/services/architect.ts (hunk 1 above)
Comment (Convly) [reply packages/core/content-type-builder/server/src/services/architect.ts:155]: I hate having to do this :grimacing: 
But if we don't have a choice, why not

File: packages/core/content-type-builder/server/src/services/architect.ts [hunk 3]
```diff
@@ -151,12 +152,46 @@ const systemPrompt = `
           </examples>
     `;
 
+const applyDefaultValues = (schema: any) => {
+  // Transform info properties to be kebab-case always
+  schema.info.singularName = schema.info.singularName.toLowerCase().replace(/\s+/g, '-');
+  schema.info.pluralName = schema.info.pluralName.toLowerCase().replace(/\s+/g, '-');
+  if (!schema.info.name) schema.info.name = schema.info.pluralName;
```
Comment (Convly) [inline packages/core/content-type-builder/server/src/services/architect.ts:159]: :eyes:

File: packages/core/content-type-builder/server/src/services/architect.ts [hunk 4]
```diff
@@ -151,12 +152,46 @@ const systemPrompt = `
...
+  return schema;
+};
+
 export const create = async (prompt: string, previousSchema?: object) => {
   console.log({ previousSchema });
 
   const { object } = await generateObject({
     model: anthropic('claude-3-sonnet-20240229', {}),
-    maxRetries: 3,
+    maxRetries: 10,
```
Comment (Convly) [inline packages/core/content-type-builder/server/src/services/architect.ts:194]: We should probably make this configurable :sweat_smile:

File: packages/core/content-type-builder/server/src/services/architect.ts [hunk 5]
```diff
@@ -166,7 +201,7 @@ export const create = async (prompt: string, previousSchema?: object) => {
         singularName: z.string(),
         pluralName: z.string(),
         description: z.string().optional(),
-        name: z.string(),
+        name: z.string().optional(), // Model forgets to add this sometimes
```
Comment (Convly) [inline packages/core/content-type-builder/server/src/services/architect.ts:204]: LLMs are dumb :man_shrugging:

Comment (Marc-Roig) [reply packages/core/content-type-builder/server/src/services/architect.ts:204]: hahaha yes they are,
we need to adapt the prompt for it to be reliable, and the best approach imo is to just ask for dumb things, and make it spit dumb plain jsons, and avoid nested keys if possible

File: packages/core/content-type-builder/server/src/services/architect.ts (hunk 3 above)
Comment (Marc-Roig) [reply packages/core/content-type-builder/server/src/services/architect.ts:159]: hehe

**Comment by Marc-Roig** [comment]
Same, this is an elegant solution but also woulnd't be the final solution, 
a much better solution would have specific repairs for every part of the schema (repair fields, info, relations, ...)

File: packages/core/content-type-builder/server/src/services/architect.ts (hunk 1 above)
Comment (Eventyret) [reply packages/core/content-type-builder/server/src/services/architect.ts:155]: Can't we cast it to unknown as that's what it is ?

Comment (Bassel17) [reply packages/core/content-type-builder/server/src/services/architect.ts:155]: we do what we have for a successful demo

File: packages/core/content-type-builder/server/src/services/architect.ts (hunk 4 above)
Comment (Eventyret) [reply packages/core/content-type-builder/server/src/services/architect.ts:194]: I agree could cast it to process.env

File: packages/core/content-type-builder/server/src/services/architect.ts (hunk 1 above)
Comment (Marc-Roig) [reply packages/core/content-type-builder/server/src/services/architect.ts:155]: 🫡 amen

File: packages/core/content-type-builder/server/src/services/architect.ts (hunk 4 above)
Comment (Marc-Roig) [reply packages/core/content-type-builder/server/src/services/architect.ts:194]: sure!

**Review by Convly — APPROVED** [review]

**Pull request merged**